keps/sig-network/1669-graceful-termination-local-external-traffic-policy/README.md

```diff
@@ -178,233 +183,91 @@ Yes, there will be unit tests in kube-proxy with the feature gate enabled and di
 
-<!--
-Describe manual testing that was done and the outcomes.
-Longer term, we may want to require automated upgrade/rollback tests, but we
-are missing a bunch of machinery and tooling and can't do that now.
--->
-
-TBD for beta.
+No, but upgrade testing should be done prior to Beta.
 
 ###### Is the rollout accompanied by any deprecations and/or removals of features, APIs, fields of API types, flags, etc.?
 
-<!--
-Even if applying deprecation policies, they may still surprise some users.
--->
-
-TBD for beta.
+No.
 
 ### Monitoring Requirements
 
-<!--
-This section must be completed when targeting beta to a release.
--->
-
 ###### How can an operator determine if the feature is in use by workloads?
 
-<!--
-Ideally, this should be a metric. Operations against the Kubernetes API (e.g.,
-checking if there are objects with field X set) may be a last resort. Avoid
-logs or events for this purpose.
--->
-
-TBD for beta.
+* A Service Type=LoadBalancer sets externalTrafficPolicy=Local.
+* The load balancer implementation uses `spec.healthCheckNodePort` for node health checking.
+* A Pod can receive traffic during termination.
 
 ###### What are the SLIs (Service Level Indicators) an operator can use to determine the health of the service?
 
-<!--
-Pick one more of these and delete the rest.
--->
-
-TBD for beta.
-
 - [ ] Metrics
   - Metric name:
   - [Optional] Aggregation method:
   - Components exposing the metric:
-- [ ] Other (treat as last resort)
-  - Details:
+- [X] Other (treat as last resort)
+  - Details: SLIs are difficult to measure for this feature since health of a service is dependant on the underlying process in the Pod as well as the load balancer implementation fronting the service.
 
 ###### What are the reasonable SLOs (Service Level Objectives) for the above SLIs?
 
-<!--
-At a high level, this usually will be in the form of "high percentile of SLI
-per day <= X". It's impossible to provide comprehensive guidance, but at the very
-high level (needs more precise definitions) those may be things like:
-  - per-day percentage of API calls finishing with 5XX errors <= 1%
-  - 99% percentile over day of absolute value from (job creation time minus expected
-    job creation time) for cron job <= 10%
-  - 99,9% of /health requests per day finish with 200 code
--->
-
-TBD for beta.
+SLOs are difficult to measure for this feature since health of a service is dependant on the underlying process in the Pod as well as the load balancer implementation fronting the service.
 
 ###### Are there any missing metrics that would be useful to have to improve observability of this feature?
 
-<!--
-Describe the metrics themselves and the reasons why they weren't added (e.g., cost,
-implementation difficulties, etc.).
--->
-
-TBD for beta.
+We may consider adding metrics for total endpoints that are in the terminating state -- this will be evaluated based on the
+cardinality of such metrics.
```

Comment:
I think I'm not fully following - can you clarify?

Reply:
This is saying that we _can_ add some level of metrics for terminating endpoints, but we need to be careful about the labels we apply to them. For example, if we included the endpoint as part of the metrics label, we are exploding metrics cardinality because every endpoint is unique. So maybe per endpoint metrics is not possible but total endpoints is. But I'm on the fence about how useful total endpoints is if you can't map it back to which nodes and pods it applies to

Reply:
regardless I'm going to take a stab at trying to add metrics for this feature, but I think it wouldn't be that useful if the metric does not surface per pod / endpoint level details.